src/Microsoft.FeatureManagement.Telemetry.ApplicationInsights/TargetingTelemetryInitializer.cs

```diff
@@ -0,0 +1,52 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+//
+
+using Microsoft.ApplicationInsights.Channel;
+using Microsoft.ApplicationInsights.DataContracts;
+using Microsoft.ApplicationInsights.Extensibility;
+using System.Diagnostics;
+
+namespace Microsoft.FeatureManagement.Telemetry.ApplicationInsights
+{
+    /// <summary>
+    /// Used to add targeting information to outgoing Application Insights telemetry.
+    /// </summary>
+    public class TargetingTelemetryInitializer : ITelemetryInitializer
+    {
+        private const string TargetingIdKey = $"Microsoft.FeatureManagement.TargetingId";
+
+        /// <summary>
+        /// Creates an instance of the TargetingTelemetryInitializer
+        /// </summary>
+        public TargetingTelemetryInitializer() { }
+
+        /// <summary>
+        /// When telemetry is initialized, adds targeting information to all relevant telemetry.
+        /// </summary>
+        /// <param name="telemetry">The <see cref="ITelemetry"/> to be initialized.</param>
+        /// <exception cref="ArgumentNullException">Thrown if the any param is null.</exception>
+        public void Initialize(ITelemetry telemetry)
+        {
+            if (telemetry == null)
+            {
+                throw new ArgumentNullException(nameof(telemetry));
+            }
+
+            // Extract the targeting id from the current activity's baggage
+            string targetingId = Activity.Current?.Baggage.FirstOrDefault(t => t.Key == TargetingIdKey).Value;
+
+            // Don't modify telemetry if there's no available targeting id
+            if (string.IsNullOrEmpty(targetingId))
+            {
+                return;
+            }
+
+            // Telemetry.Properties is deprecated in favor of ISupportProperties
+            if (telemetry is ISupportProperties telemetryWithSupportProperties)
```

Comment:
Is there a case that this wouldn't be true, perhaps if a user used an old version of app insights?

Reply:
My understanding is no, that they will always be available. The deprecation message for telemetry.Properties says:
> TelemetryContext.Properties is obsolete. Use GlobalProperties to set global level properties. For properties at item level, use ISupportProperties.Properties.

They were added in v2.21.0. .NET 6 and above should use at least v2.21.0, so I think we can assume this will be available. The versions before that are all deprecated (actually including v2.21.0 as well). v2.22.0 is the latest.